### ISSUE TITLE
False positive: unresolved reference hash

### ISSUE THREAD

**Issue by twistedfall** [intellij-rust/intellij-rust#3136]
## Environment

* **Intellij-Rust plugin version:** 0.2.0.2112-183
* **Rust toolchain version:** 1.31.0 (abe02cefd 2018-12-04)
* **IDE name and version:** IntelliJ IDEA 2018.3.1 (Ultimate Edition) Build #IU-183.4588.61, built on December 4, 2018
* **Operating system:** Gentoo

## Problem description

Hello!

I'm faced with false inspection triggering. The following valid code:
```rust
use std::hash;

struct A(u32);

impl hash::Hash for A {
	fn hash<H: hash::Hasher>(&self, state: &mut H) {
		self.0.hash(state)
	}
}
```
[rust playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=92d16b49cbbdfbf07d106c143a6a8de1)

Produces the following error:

![screenshot_20181210_131348](https://user-images.githubusercontent.com/406037/49732096-bbff9180-fc7d-11e8-8959-ac1c155416f8.png)

The actual error message is:
Unresolved reference: hash

**Comment by Undin** [comment]
Looks like it's the same bug as in #3083

**Comment by twistedfall** [comment]
Looks similar indeed. Just to add, if you import `std::hash::Hash` then the error goes away, but rustc will issue a warning about unused import.

**Comment by vlad20012** [comment]
Fixes by #3698
